lightning/src/ln/channelmanager.rs

```diff
@@ -316,6 +317,7 @@ pub(crate) struct HTLCPreviousHopData {
 	incoming_packet_shared_secret: [u8; 32],
 	phantom_shared_secret: Option<[u8; 32]>,
 	blinded_failure: Option<BlindedFailure>,
+	channel_id: Option<ChannelId>,
```

Comment:
Same here, could this be non-optional?

Reply:
Yes there are a few cases I need to go back and remove Options :) They're only required for serialization compatibility. Sorry!